nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -183,7 +183,7 @@ void ObstacleLayer::onInitialize()
       source.c_str(), topic.c_str(),
       global_frame_.c_str(), expected_update_rate, observation_keep_time);
 
-    rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
+    rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_sensor_data;
```

Comment:
I'd recommend just using the QOS as is. Storing 50 laser scans seems excessive to me.

I'd recommend deleting the `custom_qos_profile` variable entirely and just passing `rmw_qos_profile_sensor_data` in directly below.